x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java

```diff
@@ -198,6 +196,13 @@ private void cleanupSettings() throws IOException {
         final Request request = new Request("PUT", "_cluster/settings");
         request.setJsonEntity(Strings.toString(builder));
         client().performRequest(request);
+
+        // await the cleanup to generate deprecation too. It will be removed in the @Before
+        assertBusy(() -> {
+            List<Map<String, Object>> documents = DeprecationTestUtils.getIndexedDeprecations(client());
+            logger.warn(documents);
+            assertThat(documents, hasSize(2));
```

Comment:
Will the deprecation log still contain the messages from the test, or does `getIndexedDeprecations` only get new log entries?

Reply:
turns out the cleanup should not be able to generate more deprecations.
it uses the same settings which previously generated a deprecation. Meaning it will be throttled (in a ratelimitingfilter) and would not generate new deprecations